pydra/engine/tests/test_workflow.py

```diff
@@ -1438,6 +1632,68 @@ def test_wf_ndstate_cachelocations(plugin, tmpdir):
     assert not wf2.output_dir.exists()
 
 
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_ndstate_cachelocations_updatespl(plugin, tmpdir):
+    """
+    Two wfs with identical inputs and node state (that is set after adding the node!);
+    the second wf has cache_locations and should not recompute the results
+    """
+    cache_dir1 = tmpdir.mkdir("test_wf_cache3")
+    cache_dir2 = tmpdir.mkdir("test_wf_cache4")
+
+    wf1 = Workflow(name="wf", input_spec=["x", "y"], cache_dir=cache_dir1)
+    wf1.add(
+        multiply(name="mult", x=wf1.lzin.x, y=wf1.lzin.y).split(splitter=("x", "y"))
+    )
+    wf1.add(add2_wait(name="add2", x=wf1.mult.lzout.out))
+    wf1.set_output([("out", wf1.add2.lzout.out)])
+    wf1.inputs.x = [2, 20]
+    wf1.inputs.y = [3, 4]
+    wf1.plugin = plugin
+
+    t0 = time.time()
+    with Submitter(plugin=plugin) as sub:
+        sub(wf1)
+    t1 = time.time() - t0
+
+    results1 = wf1.result()
+    assert results1.output.out == [8, 82]
+
+    wf2 = Workflow(
+        name="wf",
+        input_spec=["x", "y"],
+        cache_dir=cache_dir2,
+        cache_locations=cache_dir1,
+    )
+    wf2.add(multiply(name="mult", x=wf2.lzin.x, y=wf2.lzin.y))
+
+    wf2.add(add2_wait(name="add2", x=wf2.mult.lzout.out))
+    wf2.mult.split(splitter=("x", "y"))
+    wf2.set_output([("out", wf2.add2.lzout.out)])
+    wf2.inputs.x = [2, 20]
+    wf2.inputs.y = [3, 4]
+    wf2.plugin = plugin
+
+    t0 = time.time()
+    with Submitter(plugin=plugin) as sub:
+        sub(wf2)
+    t2 = time.time() - t0
+
+    results2 = wf2.result()
+    assert results2.output.out == [8, 82]
+
+    # checking execution time
+    assert t1 > 3
+    assert t2 < 0.1
```

Comment:
i'm going to merge this, but the way i think we should evaluate whether the correct tasks are rerun is using messaging. (see #9)

Reply:
are you sure that you pointed me to the right pr?

Reply:
oops - updated the link

Reply:
yes, we should add it, but just fyi, the time is not the only check in the tests, I'm also checking if the outputdir is created.